lib/jekyll/theme.rb

```diff
@@ -43,7 +47,7 @@ def path_for(folder)
     end
 
     def realpath_for(folder)
-      File.realpath(Jekyll.sanitized_path(root, "_#{folder}"))
+      File.realpath(Jekyll.sanitized_path(root, folder.to_s))
```

Comment:
If we're passing symbols like `:_layouts`, why is it called `realpath_for`? It'd be something like `in_theme_dir` which makes more sense. I think `realpath_for` should be an abstraction (special case `:assets`, for example)